From 659526d33b2db1b0924b2769b6a0b824709860cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludovic=20Court=C3=A8s?= <ludo@gnu.org>
Date: Mon, 18 Nov 2019 15:14:56 +0100
Subject: [PATCH 2/3] ports: 'scm_port_poll' honors "w" flags.

Fixes <https://bugs.gnu.org/36709>.
Reported by Mark H Weaver <mhw@netris.org>.

* libguile/ports.c (scm_port_poll): Replace second POLLIN by POLLOUT.
---
 libguile/ports.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libguile/ports.c b/libguile/ports.c
index 45e62f4e4..0084f0c37 100644
--- a/libguile/ports.c
+++ b/libguile/ports.c
@@ -1440,7 +1440,7 @@ SCM_DEFINE (scm_port_poll, "port-poll", 2, 1, 0,
   if (scm_i_string_contains_char (events, '!'))
     c_events |= POLLPRI;
   if (scm_i_string_contains_char (events, 'w'))
-    c_events |= POLLIN;
+    c_events |= POLLOUT;
 
   scm_dynwind_begin (0);
   scm_dynwind_acquire_port (port);
-- 
2.24.1

